### PR TITLE
feat: Pattern - Azure Image Builder - `avm/ptn/virtual-machine-images/azure-image-builder`

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

This PR introduces the Azure Image Builder pattern to AVM

Depends on
- #3006 
- #3007 
- #2992 
- #2756

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.ptn.virtual-machine-images.azure-image-builder](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.ptn.virtual-machine-images.azure-image-builder.yml)      |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

**Comment by eriqua** [comment]
Hey @AlexanderSehr, as all PR mentioned in the description are closed, wondering if this is ready for review (conflicts aside)

File: .github/workflows/avm.ptn.virtual-machine-images.azure-image-builder.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        description: "Remove deployed module"
+        required: false
+        default: true
+      customLocation:
+        type: string
+        description: "Default location overwrite (e.g., eastus)"
+        required: false
+  push:
+    # branches:
+    #   - main
```
Comment (AlexanderSehr) [inline .github/workflows/avm.ptn.virtual-machine-images.azure-image-builder.yml:27]: Re-enable once done

**Comment by AlexanderSehr** [comment]
> Hey @AlexanderSehr, as all PR mentioned in the description are closed, wondering if this is ready for review (conflicts aside)

Fundamentally, yes. Just running another test now that all other changes got merged :)

File: avm/ptn/virtual-machine-images/azure-image-builder/README.md [hunk 2]
```diff
@@ -0,0 +1,1159 @@
...
+
+## Notes
+
+
+### Prerequisites
+
+The deployments described in the following sections assume certain prerequisites to be in place prior to deployment.
+
+- The deployment principal (e.g., the Service Principal tied to the deploying Service Connection) must have at least `Contributor` & `User Access Adminitrator` permissions on the target subscription to be able to deploy both resources and assign permissions to created user-assigned identities
+- IF you have a policy in place that prevents Storage Accounts from being deployed without a Firewall, you have to create an exemption for the Image Template / Staging Resource Group you can configure for the Image Template Resource (parameter `imageTemplateResourceGroupName`). The rationale is that the Azure-Image-Builder service uses this resource group to deploy both temporal resources used during the image build (e.g., a Virtual Machine), as well as a Storage Account to store temporal files & a 'packerlogs/customization.log' file in (which contains the logs of the image build). This Storage Account has no firewall configured, has a random name, and cannot be configured at deploy time.
```
Comment (rahalan) [inline avm/ptn/virtual-machine-images/azure-image-builder/README.md:1044]:
```suggestion
- If you have a policy in place that prevents Storage Accounts from being deployed without a Firewall, you have to create an exemption for the Image Template / Staging Resource Group you can configure for the Image Template Resource (parameter `imageTemplateResourceGroupName`). The rationale is that the Azure-Image-Builder service uses this resource group to deploy both temporal resources used during the image build (e.g., a Virtual Machine), as well as a Storage Account to store temporal files & a 'packerlogs/customization.log' file in (which contains the logs of the image build). This Storage Account has no firewall configured, has a random name, and cannot be configured at deploy time.
```

**Review by rahalan — APPROVED** [review]

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**